src/ui/pages/grid/Grid.js

```diff
@@ -1,14 +1,54 @@
-import React from 'react';
-import queryString from "query-string";
 import EAMGrid from "eam-components/dist/ui/components/eamgrid";
+import queryString from "query-string";
+import React from 'react';
+import { Link } from 'react-router-dom';
+import GridTools from '../../../tools/GridTools';
+import Typography from '@material-ui/core/Typography';
+
+const treatParamAsList = (param) => (param === undefined || param === null) ? []
+    : Array.isArray(param) ? param
+    : param.includes(",") ? param.split(",")
+    : [param]
+
+
+const cellRenderer = ({assetColumns, workorderColumns, locationColumns, partColumns}) => (cell, row) => {
+    return treatParamAsList(assetColumns).includes(cell.t)  ?
+            getLink("/equipment/", cell.value)
+        : treatParamAsList(locationColumns).includes(cell.t)  ?
+            getLink("/location/", cell.value)
+        : treatParamAsList(workorderColumns).includes(cell.t)  ?
+            getLink("/workorder/", cell.value)
+        : treatParamAsList(partColumns).includes(cell.t)  ?
+            getLink("/part/", cell.value)
+        : ["equipmentno", "obj_code", "evt_object", , "equipment"].includes(cell.t) ?
+            getLink("/equipment/", cell.value)
+        : ["location"].includes(cell.t) ?
+            getLink("/location/", cell.value)
+        : ["workordernum", "evt_code", "parentwo"].includes(cell.t) ?
+            getLink("/workorder/", cell.value)
+        : ["part"].includes(cell.t) ?
+            getLink("/part/", cell.value)
+        : null
+        ;
+}
```

Comment:
I feel this is a bit repetitive, but the logic itself is very clear, perhaps it is a good choice to have it this way here.

Do you think something like the **untested** code below would be an improvement?
```
const cellRenderer = userColumns => (cell, row) => {
    const userColumnToType = {
        assetColumns: 'equipment',
        locationColumns: 'location',
        workorderColumns: 'workorder',
        partColumns: 'part',
    };

    const typeToDefaultColumns = {
        equipment: ['equipmentno', 'obj_code', 'evt_object', 'equipment'],
        location: ['location'],
        workorder: ['workordernum', 'evt_code', 'parentwo'],
        part: ['part'],
    }

    const link = userColumnToType[Object.keys(userColumnToType)
        .find(userColumn => treatParamAsList(userColumns[userColumn]).includes(cell.t))]
        || Object.keys(typeToDefaultColumns)
        .find(type => typeToDefaultColumns[type].includes(cell.t));

    return link === undefined ? null : getLink(`/${link}/`, cell.value);
}
```

Reply:
This piece of code was actually from another feature/issue, but yes, your suggestion is an improvement, thank you